src/modules/order/completeOrder.ts

```diff
@@ -1,20 +1,45 @@
+import assert from 'assert';
+import { assign } from 'lodash';
 import { HttpModule } from '../HttpModule';
 
-const completeOrder: HttpModule<{ status: number }> = {
-  authorise({ $roles }, { user }) {
-    return $roles.validate(user, []);
-  },
-  exec() {
-    //   events: [
-    //     productsConfirmedToOrder,
-    //     paymentDetailsAddedToContact,
-    //     paymentDetailsAddedToOrder,
-    //     paymentProcessedThroughStripe,
-    //     giftCardUsed,
-    //     checkoutCompleted,
-    //     confirmationEmailQueued,
-    // ]
-    return Promise.resolve({ status: 200 });
+const completeOrder: HttpModule<{ orderId: string; paymentIntent: string }, void> = {
+  exec({ $database, createEvent }, { orderId, paymentIntent }) {
+    assert(orderId, 'Order Id not found');
+    assert(paymentIntent, 'Payment Intent not provided');
+    return $database
+      .then(({ document, startTransaction }) => {
+        return document.fetch('orders', orderId).then(loadedOrder => {
+          assert(loadedOrder, 'Order not found');
+          assert(loadedOrder.orderStatus, 'Order already completed.');
+
+          assign(loadedOrder, {
+            ...loadedOrder,
+            paymentIntent,
+            orderStatus: 'completed',
+          });
+
+          const events = [
+            createEvent('OrderPaymentProcessedThroughStripe', {
```

Comment:
Does storing the intent and completing the order happen at the same time?

Reply:
Correct We have already charged the card